src/Instrumentation/AwsSdk/AwsSdkInstrumentation.php

```diff
@@ -37,8 +44,81 @@ public function init(): bool
         return true;
     }
 
+    public function setPropagator(TextMapPropagatorInterface $propagator): void
+    {
+        $this->propagator = $propagator;
+    }
+
+    public function getPropagator(): TextMapPropagatorInterface
+    {
+        return $this->propagator;
+    }
+
+    public function setTracerProvider(TracerProviderInterface $tracerProvider): void
+    {
+        $this->tracerProvider = $tracerProvider;
+    }
+
+    public function getTracerProvider(): TracerProviderInterface
+    {
+        return $this->tracerProvider;
+    }
+
+    public function getTracer(): TracerInterface
+    {
+        return $this->tracerProvider->getTracer('io.opentelemetry.contrib.php');
+    }
+
     public function activate(): bool
     {
+        AwsGlobal::setInstrumentation($this);
+
+        try {
+            runkit7_method_copy('Aws\AwsClient', '__call_copy', 'Aws\AwsClient', '__call');
+            runkit7_method_copy('Aws\AwsClient', 'executeAsync_copy', 'Aws\AwsClient', 'executeAsync');
+
+            runkit7_method_redefine(
+                'Aws\AwsClient',
+                '__call',
+                '$name, $args',
+                '   
+                
+                $tracer = \OpenTelemetry\Instrumentation\AwsSdk\AwsGlobal::getInstrumentation()->getTracer();
+                $carrier = [];
+                
+                $this->span = $tracer->spanBuilder($this->getApi()->getServiceName() . "." . $name)->setSpanKind(\OpenTelemetry\Instrumentation\AwsSdk\AwsSdkInstrumentation::SPAN_KIND)->startSpan();
+                $this->scope = $this->span->activate();
+                
+                $propagator = \OpenTelemetry\Instrumentation\AwsSdk\AwsGlobal::getInstrumentation()->getPropagator();
+                $propagator->inject($carrier);
+                
+                $this->span->setAttributes([
+                    "rpc.method" => $name,
+                    "rpc.service" => $this->getApi()->getServiceName(),
+                    "rpc.system" => "aws-api",
+                    "aws.region" => $this->getRegion()
+                ]);
+                
+                return $this->__call_copy($name, $args);
+                    
+                
+                ',
+            );
+
+            runkit7_method_redefine(
+                'Aws\AwsClient',
+                'executeAsync',
+                '$command',
+                ' 
+                    $this->scope->detach();
```

Comment:
Should likely be moved to `__call`, after `$this->__call_copy($name, $args);`?
Doesn't detach scope for non-`...Async` methods and doesn't handle concurrent calls.